apps/web/src/components/RecruitmentProcesses/OffPlatformProcessDialog.tsx

```diff
@@ -0,0 +1,527 @@
+  uiMessages,
+} from "@gc-digital-talent/i18n";
+import {
+  CreateOffPlatformRecruitmentProcessInput,
+  FragmentType,
+  getFragment,
+  graphql,
+  HiringPlatform,
+  OffPlatformRecruitmentProcess,
+  UpdateOffPlatformRecruitmentProcessInput,
+} from "@gc-digital-talent/graphql";
+import {
+  Combobox,
+  Input,
+  localizedEnumToOptions,
+  objectsToSortedOptions,
+  RadioGroup,
+  Select,
+} from "@gc-digital-talent/forms";
+import { unpackMaybes } from "@gc-digital-talent/helpers";
+
+import processMessages from "~/messages/processMessages";
+import { splitAndJoin } from "~/utils/nameUtils";
+import { getClassificationName } from "~/utils/poolUtils";
+
+export const OffPlatformProcessDialog_Fragment = graphql(/* GraphQL */ `
+  fragment OffPlatformProcessDialog on Query {
+    me {
+      id
+    }
+    departments {
+      id
+      name {
+        en
+        fr
+      }
+    }
+    classifications {
+      id
+      name {
+        localized
+      }
+      group
+      level
+    }
+    hiringPlatforms: localizedEnumStrings(enumName: "HiringPlatform") {
+      value
+      label {
+        en
+        fr
+      }
+    }
+  }
+`);
+
+const CreateOffPlatformProcess_Mutation = graphql(/* GraphQL */ `
+  mutation createOffPlatformRecruitmentProcess(
+    $process: CreateOffPlatformRecruitmentProcessInput!
+  ) {
+    createOffPlatformRecruitmentProcess(process: $process) {
+      id
+    }
+  }
+`);
+
+const UpdateOffPlatformProcess_Mutation = graphql(/* GraphQL */ `
+  mutation updateOffPlatformRecruitmentProcess(
+    $process: UpdateOffPlatformRecruitmentProcessInput!
+  ) {
+    updateOffPlatformRecruitmentProcess(process: $process) {
+      id
+    }
+  }
+`);
+
+const DeleteOffPlatformProcess_Mutation = graphql(/* GraphQL */ `
+  mutation deleteOffPlatformRecruitmentProcess($id: UUID!) {
+    deleteOffPlatformRecruitmentProcess(id: $id) {
+      id
+    }
+  }
+`);
+
+interface FormValues {
+  processNumber: string | null | undefined;
+  department: string;
+  classificationGroup: string;
+  classificationLevel: string;
+  platform: HiringPlatform;
+  platformOther: string | null | undefined;
+}
+
+interface OffPlatformProcessDialogProps {
+  query?: FragmentType<typeof OffPlatformProcessDialog_Fragment>;
+  process?: Omit<OffPlatformRecruitmentProcess, "user">;
+}
+
+const OffPlatformProcessDialog = ({
+  query,
+  process,
+}: OffPlatformProcessDialogProps) => {
+  const intl = useIntl();
+  const [open, setOpen] = useState(false);
+  const methods = useForm<FormValues>({
+    defaultValues: {
+      processNumber: process?.processNumber,
+      department: process?.department?.id,
+      classificationGroup: process?.classification?.group,
+      classificationLevel: process?.classification?.level?.toString(),
+      platform: process?.platform?.value,
+      platformOther: process?.platformOther,
+    },
+  });
+  const { watch, handleSubmit, resetField, reset } = methods;
+  const [groupSelection, platformSelection] = watch([
```

Comment:
Feel free to ignore, but you could get rid of this classification stuff if you were able to reuse the apps/web/src/components/ClassificationInput/ClassificationInput.tsx component.

Reply:
That's definitely nicer - done in [3e47477](https://github.com/GCTC-NTGC/gc-digital-talent/pull/14369/commits/3e47477b3f05f882f840d6b80643401e7063a458)